### PR TITLE
Feature/vr 4592 dataset blobs description metadata

### PR DESCRIPTION
**Pull request by evgeniilezhnin**

### CODE REVIEW

File: backend/src/main/java/ai/verta/modeldb/entities/metadata/LabelsMappingDescriptionEntity.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import ai.verta.modeldb.metadata.IdentificationType;
+import ai.verta.modeldb.metadata.VersioningCompositeIdentifier;
+import ai.verta.modeldb.utils.ModelDBUtils;
+import java.io.Serializable;
+import java.util.Objects;
+import javax.persistence.*;
+
+@Entity
+@Table(name = "labels_mapping_description")
+public class LabelsMappingDescriptionEntity implements LabelsMappingEntityBase {
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/entities/metadata/LabelsMappingDescriptionEntity.java]: name change of DescriptionMappingEntity

File: backend/src/main/java/ai/verta/modeldb/entities/metadata/LabelsMappingEntityBase.java [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package ai.verta.modeldb.entities.metadata;
+
+public interface LabelsMappingEntityBase {
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/entities/metadata/LabelsMappingEntityBase.java]: the previous class was actually storing Label. A better naming in my option would be MetadataMappingEntityBase, with getMetadata

File: backend/src/main/java/ai/verta/modeldb/metadata/MetadataDAORdbImpl.java [hunk 3]
```diff
@@ -27,6 +33,17 @@
           .append(ModelDBConstants.ENTITY_TYPE)
           .append(" = :entityType")
           .toString();
+  private static final String GET_LABELS_DESCRIPTION_HQL =
+      new StringBuilder("From LabelsMappingDescriptionEntity lm where lm.id.")
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/metadata/MetadataDAORdbImpl.java]: name change will affect this

File: backend/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java [hunk 4]
```diff
@@ -26,12 +26,7 @@
 import ai.verta.uac.GetCollaboratorResponse;
 import ai.verta.uac.ShareViaEnum;
 import ai.verta.uac.UserInfo;
-import com.google.protobuf.Any;
-import com.google.protobuf.GeneratedMessageV3;
-import com.google.protobuf.InvalidProtocolBufferException;
-import com.google.protobuf.Message;
-import com.google.protobuf.MessageOrBuilder;
-import com.google.protobuf.Value;
+import com.google.protobuf.*;
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java]: we keep undoing , redoing these changes , if these are automated can both @evgeniilezhnin  and @ad-47  get to a consensus and update their editor settings.
I like expanding the *.

File: backend/src/main/resources/liquibase/create-versioning-tables-changelog2.xml [hunk 5]
```diff
@@ -383,4 +383,23 @@
 			<column name="repository_id" type="int8"/>
 		</addColumn>
 	</changeSet>
+	<changeSet id="add-description-blob-table" author="lezhevg">
+		<preConditions onFail="MARK_RAN">
+			<not>
+				<tableExists tableName="labels_mapping_description"/>
```
Comment (ravishetye) [inline backend/src/main/resources/liquibase/create-versioning-tables-changelog2.xml]: description_mapping

File: backend/src/main/java/ai/verta/modeldb/metadata/MetadataDAORdbImpl.java [hunk 6]
```diff
@@ -83,15 +93,83 @@ public boolean addLabels(IdentificationType id, List<String> labels) {
...
+
   @Override
   public List<String> getLabels(IdentificationType id) {
     try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
-      Query query = session.createQuery(GET_LABELS_HQL);
-      query.setParameter("entityHash", getEntityHash(id));
-      query.setParameter("entityType", id.getIdTypeValue());
-      List<LabelsMappingEntity> labelsMappingEntities = query.list();
+      List<LabelsMappingEntityBase> labelsMappingEntities;
+      if (id.getIdType() == VERSIONING_REPO_COMMIT_BLOB_DESCRIPTION) {
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/metadata/MetadataDAORdbImpl.java]: this if should be on what metadata is being processed , either label or description.

**Review by ravishetye — CHANGES_REQUESTED** [review]

File: backend/src/main/java/ai/verta/modeldb/metadata/MetadataDAORdbImpl.java (hunk 6 above)
Comment (ravishetye) [reply backend/src/main/java/ai/verta/modeldb/metadata/MetadataDAORdbImpl.java]: this function could change to List<String> getMetdata(IdentificationType id, MetadataType mt) and the if can be 
```
if(mt = MetadataType.Label){
...}
else if(mt = MetadataType.Description){
...
}

```

MetadataType could be internal enum

File: backend/src/main/java/ai/verta/modeldb/entities/metadata/LabelsMappingDescriptionEntity.java (hunk 1 above)
Comment (evgeniilezhnin) [reply backend/src/main/java/ai/verta/modeldb/entities/metadata/LabelsMappingDescriptionEntity.java]: changed to metadata_parameter_mapping

File: backend/src/main/java/ai/verta/modeldb/entities/metadata/LabelsMappingEntityBase.java (hunk 2 above)
Comment (evgeniilezhnin) [reply backend/src/main/java/ai/verta/modeldb/entities/metadata/LabelsMappingEntityBase.java]: this class is not needed anymore

File: backend/src/main/java/ai/verta/modeldb/metadata/MetadataDAORdbImpl.java (hunk 3 above)
Comment (evgeniilezhnin) [reply backend/src/main/java/ai/verta/modeldb/metadata/MetadataDAORdbImpl.java]: fixed

File: backend/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java (hunk 4 above)
Comment (evgeniilezhnin) [reply backend/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java]: sorry, I was reinstalling idea recently and forgot to set this parameter to expand *. Now I think I have fixed it

File: backend/src/main/resources/liquibase/create-versioning-tables-changelog2.xml (hunk 5 above)
Comment (evgeniilezhnin) [reply backend/src/main/resources/liquibase/create-versioning-tables-changelog2.xml]: fixed

File: backend/src/main/java/ai/verta/modeldb/metadata/MetadataDAORdbImpl.java (hunk 6 above)
Comment (evgeniilezhnin) [reply backend/src/main/java/ai/verta/modeldb/metadata/MetadataDAORdbImpl.java]: now it's in separate methods

File: backend/src/main/java/ai/verta/modeldb/entities/metadata/MetadataParameterMappingEntity.java [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+import java.util.Objects;
+import javax.persistence.Column;
+import javax.persistence.Embeddable;
+import javax.persistence.EmbeddedId;
+import javax.persistence.Entity;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "metadata_parameter_mapping")
+public class MetadataParameterMappingEntity {
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/entities/metadata/MetadataParameterMappingEntity.java]: I think Property would be a better name than Parameter

File: backend/src/main/java/ai/verta/modeldb/metadata/MetadataDAO.java [hunk 8]
```diff
@@ -1,11 +1,22 @@
 package ai.verta.modeldb.metadata;
 
 import java.util.List;
+import org.hibernate.Session;
 
 public interface MetadataDAO {
   boolean addLabels(IdentificationType id, List<String> labels);
 
+  void addParameter(Session session, IdentificationType id, String key, String value);
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/metadata/MetadataDAO.java]: Property

File: backend/src/main/java/ai/verta/modeldb/metadata/MetadataDAORdbImpl.java [hunk 9]
```diff
@@ -83,16 +82,53 @@ public boolean addLabels(IdentificationType id, List<String> labels) {
     }
   }
 
+  @Override
+  public boolean addParameter(IdentificationType id, String key, String value) {
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/metadata/MetadataDAORdbImpl.java]: addProperty

**Review by ravishetye — COMMENTED** [review]
logic looks good just I would prefer calling it Property instead of Parameter.
You can merge this in if you make this change, else we can discuss on slack as a follow up.

File: backend/src/main/java/ai/verta/modeldb/entities/metadata/MetadataParameterMappingEntity.java (hunk 7 above)
Comment (evgeniilezhnin) [reply backend/src/main/java/ai/verta/modeldb/entities/metadata/MetadataParameterMappingEntity.java]: replaced

File: backend/src/main/java/ai/verta/modeldb/metadata/MetadataDAO.java (hunk 8 above)
Comment (evgeniilezhnin) [reply backend/src/main/java/ai/verta/modeldb/metadata/MetadataDAO.java]: replaced

File: backend/src/main/java/ai/verta/modeldb/metadata/MetadataDAORdbImpl.java (hunk 9 above)
Comment (evgeniilezhnin) [reply backend/src/main/java/ai/verta/modeldb/metadata/MetadataDAORdbImpl.java]: replaced

**Pull request merged**